Core.DomainModel/Organization/StsOrganizationConnection.cs

```diff
@@ -1,38 +1,141 @@
     {
+        public StsOrganizationConnection()
+        {
+            StsOrganizationChangeLogs = new List<StsOrganizationChangeLog>();
+        }
+
         public int OrganizationId { get; set; }
         public virtual Organization Organization { get; set; }
         public bool Connected { get; set; }
         /// <summary>
         /// Determines the optional synchronization depth used during synchronization from STS Organisation
         /// </summary>
         public int? SynchronizationDepth { get; set; }
+        public virtual ICollection<StsOrganizationChangeLog> StsOrganizationChangeLogs { get; set; }
+
 
         public bool SubscribeToUpdates { get; set; }
-        //TODO https://os2web.atlassian.net/browse/KITOSUDV-3317 adds the change logs here
+
         public DisconnectOrganizationFromOriginResult Disconnect()
         {
             var organizationUnits = Organization.OrgUnits.Where(x => x.Origin == OrganizationUnitOrigin.STS_Organisation).ToList();
             organizationUnits.ForEach(unit => unit.ConvertToNativeKitosUnit());
+            var removedLogs = RemoveAllLogs();
 
             Connected = false;
             SubscribeToUpdates = false;
             SynchronizationDepth = null;
-            return new DisconnectOrganizationFromOriginResult(organizationUnits);
+            return new DisconnectOrganizationFromOriginResult(organizationUnits, removedLogs);
+        }
+
+        public void Connect()
+        {
+            Connected = true;
+        }
+
+        public Maybe<OperationError> Subscribe()
+        {
+            if (!Connected)
+                return new OperationError("Organization isn't connected to the sts service", OperationFailure.BadState);
+
+            SubscribeToUpdates = true;
+            return Maybe<OperationError>.None;
+        }
+
+        public Maybe<OperationError> Unsubscribe()
+        {
+            if(!Connected)
+                return new OperationError("Organization isn't connected to the sts service", OperationFailure.BadState);
+
+            SubscribeToUpdates = false;
+            return Maybe<OperationError>.None;
+        }
+
+        public Maybe<OperationError> UpdateSynchronizationDepth(int? synchronizationDepth)
+        {
+            if (!Connected)
+                return new OperationError("Organization isn't connected to the sts service", OperationFailure.BadState);
+
+            SynchronizationDepth = synchronizationDepth;
+            return Maybe<OperationError>.None;
         }
 
         public IExternalOrganizationalHierarchyUpdateStrategy GetUpdateStrategy()
         {
             return new StsOrganizationalHierarchyUpdateStrategy(Organization);
         }
+
+        public ExternalConnectionAddNewLogsResult AddNewLog(ExternalConnectionAddNewLogInput newLogInput)
+        {
+            var newLogEntries = newLogInput.Entries.Select(x =>
```

Comment:
Is it valid to add a log if not connected?

Reply:
hint: Return type should be `Result<ExternalConnectionAddNewLogsResult, OperationError>` and client should handle it of course